rules/typescript/extensions.js

```diff
@@ -0,0 +1,207 @@
+    'init-declarations': 'off',
+    '@typescript-eslint/init-declarations': suggestions['init-declarations'],
+
+    // Enforce consistent spacing before and after keywords
+    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/keyword-spacing.md
+    'keyword-spacing': 'off',
+    '@typescript-eslint/keyword-spacing': layoutFormatting['keyword-spacing'],
+
+    // Require or disallow an empty line between class members
+    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/lines-between-class-members.md
+    'lines-between-class-members': 'off',
+    '@typescript-eslint/lines-between-class-members': layoutFormatting['lines-between-class-members'],
+
+    // Disallow generic Array constructors
+    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-array-constructor.md
+    '@typescript-eslint/no-array-constructor': 'error',
+
+    // Disallow duplicate class members
+    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-dupe-class-members.md
+    'no-dupe-class-members': 'off',
+    '@typescript-eslint/no-dupe-class-members': possibleProblems['no-dupe-class-members'],
+
+    // Disallow duplicate imports
+    // https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/no-duplicate-imports.md
+    'no-duplicate-imports': 'off',
+    '@typescript-eslint/no-duplicate-imports': possibleProblems['no-duplicate-imports'],
+
+    // Disallow Empty Functions
+    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-empty-function.md
+    'no-empty-function': 'off',
+    '@typescript-eslint/no-empty-function': suggestions['no-empty-function'],
+
+    // Disallow unnecessary parentheses
+    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-extra-parens.md
+    'no-extra-parens': 'off',
+    '@typescript-eslint/no-extra-parens': layoutFormatting['no-extra-parens'],
+
+    // Disallow unnecessary semicolons
+    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-extra-semi.md
+    'no-extra-semi': 'off',
+    '@typescript-eslint/no-extra-semi': suggestions['no-extra-semi'],
+
+    // Disallow the use of eval()-like methods
+    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-implied-eval.md
+    '@typescript-eslint/no-implied-eval': 'error',
+
+    // Disallow this keywords outside of classes or class-like objects
+    // https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/docs/rules/no-invalid-this.md
+    'no-invalid-this': 'off',
+    '@typescript-eslint/no-invalid-this': layoutFormatting['no-invalid-this'],
```

Comment:
There is something wrong with this line because ESLint currently fails with the following error:

```
$ eslint --ext .js,.ts *.js scripts packages integration

Oops! Something went wrong! :(

ESLint: 7.32.0

Error: .eslintrc.json » @vaadin/eslint-config-vaadin/typescript » /Users/vursen/dev/vaadin/web-components/node_modules/@vaadin/eslint-config-vaadin/rules/typescript/extensions.js:
	Configuration for rule "@typescript-eslint/no-invalid-this" is invalid:
	Severity should be one of the following: 0 = off, 1 = warn, 2 = error (you passed 'undefined').

    at ConfigValidator.validateRuleOptions (/Users/vursen/dev/vaadin/web-components/node_modules/@eslint/eslintrc/lib/shared/config-validator.js:144:23)
    at /Users/vursen/dev/vaadin/web-components/node_modules/@eslint/eslintrc/lib/shared/config-validator.js:199:18
    at Array.forEach (<anonymous>)
    at ConfigValidator.validateRules (/Users/vursen/dev/vaadin/web-components/node_modules/@eslint/eslintrc/lib/shared/config-validator.js:196:34)
    at ConfigValidator.validateConfigArray (/Users/vursen/dev/vaadin/web-components/node_modules/@eslint/eslintrc/lib/shared/config-validator.js:322:18)
    at CascadingConfigArrayFactory._finalizeConfigArray (/Users/vursen/dev/vaadin/web-components/node_modules/@eslint/eslintrc/lib/cascading-config-array-factory.js:493:23)
    at CascadingConfigArrayFactory.getConfigArrayForFile (/Users/vursen/dev/vaadin/web-components/node_modules/@eslint/eslintrc/lib/cascading-config-array-factory.js:299:21)
    at FileEnumerator._iterateFilesWithFile (/Users/vursen/dev/vaadin/web-components/node_modules/eslint/lib/cli-engine/file-enumerator.js:365:43)
    at FileEnumerator._iterateFiles (/Users/vursen/dev/vaadin/web-components/node_modules/eslint/lib/cli-engine/file-enumerator.js:346:25)
    at FileEnumerator.iterateFiles (/Users/vursen/dev/vaadin/web-components/node_modules/eslint/lib/cli-engine/file-enumerator.js:296:59)
```